helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -172,13 +173,15 @@
   private boolean _inMaintenanceMode;
 
   /**
-   * The timer that can periodically run the rebalancing pipeline. The timer will start if there is
-   * one resource group has the config to use the timer.
+   * The executors that can periodically run the rebalancing pipeline. A
+   * SingleThreadScheduledExecutor will start if there is resource group that has the config to do
+   * periodically rebalance.
    */
-  Timer _periodicalRebalanceTimer = null;
+  private static final ScheduledExecutorService _periodicalRebalanceExecutor =
+      Executors.newSingleThreadScheduledExecutor();
```

Comment:
Good suggestion. Will update.

Reply:
I think this is a valid suggestion. However, I did not find setRemoveOnCancelPolicy() for interface ScheduledExecutorService  we are using here..

Reply:
This method is only available in ScheduledThreadPoolExecutor. Is it easy to switch to ScheduledThreadPoolExecutor?

Reply:
Thanks Molly! We had an offline discussion.  Since cancelled tasks will be removed from the work queue when delay elapses, and there is no setRemoveOnCancelPolicy() for interface ScheduledExecutorService. We think for our current workload, it is ok to keep the way it is now.